src/actions/videoEdit/ConcatenateAction.ts

```diff
@@ -124,6 +133,28 @@ class ConcatenateAction extends Action {
       close
     ].filter((a) => a).join('/');
   }
+
+  static fromJson(actionModel: IActionModel, transformationFromJson: ITransformationFromJson): ConcatenateAction {
+    const {source, transition, prepend, duration} = (actionModel as IConcatenateActionModel);
+
+    // We are using this() to allow inheriting classes to use super.fromJson.apply(this, [actionModel])
+    // This allows the inheriting classes to determine the class to be created
+    // @ts-ignore
+    const result = new this(VideoSource.fromJson(source, transformationFromJson));
```

Comment:
For clarity, can we replace `new this()` with `new ConcatenateAction`?

I'm pretty sure nothing extends `ConcatenateAction` (so we don't need to have a very generic solution.

I also don't understand the comment, what inheriting classes are you referring to here?

Reply:
we have a "new this" in every fromJson, the comment is copy pasted in all of them for any newcomers to understand why we chose to do that. even if nothing extends it now, it might be extended. also, this technique is used by some inheriting resize actions for example, so we keep it aligned.